connector/signaltometricsconnector/connector_test.go

```diff
@@ -0,0 +1,355 @@
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package signaltometricsconnector
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	"github.com/elastic/opentelemetry-collector-components/connector/signaltometricsconnector/config"
+	"github.com/elastic/opentelemetry-collector-components/connector/signaltometricsconnector/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/confmaptest"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/connector/connectortest"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest"
+)
+
+const testDataDir = "testdata"
+
+func TestConnectorWithTraces(t *testing.T) {
+	testCases := []string{
+		"sum",
+		"histograms",
+		"exponential_histograms",
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for _, tc := range testCases {
+		t.Run(tc, func(t *testing.T) {
+			traceTestDataDir := filepath.Join(testDataDir, "traces")
+			inputTraces, err := golden.ReadTraces(filepath.Join(traceTestDataDir, "traces.yaml"))
+			require.NoError(t, err)
+
+			next := &consumertest.MetricsSink{}
+			tcTestDataDir := filepath.Join(traceTestDataDir, tc)
+			factory, settings, cfg := setupConnector(t, tcTestDataDir)
+			connector, err := factory.CreateTracesToMetrics(ctx, settings, cfg, next)
+			require.NoError(t, err)
+			require.IsType(t, &signalToMetrics{}, connector)
+			expectedMetrics, err := golden.ReadMetrics(filepath.Join(tcTestDataDir, "output.yaml"))
+			require.NoError(t, err)
+
+			require.NoError(t, connector.ConsumeTraces(ctx, inputTraces))
+			require.Len(t, next.AllMetrics(), 1)
+			assertAggregatedMetrics(t, expectedMetrics, next.AllMetrics()[0])
+		})
+	}
+}
+
+func TestConnectorWithMetrics(t *testing.T) {
+	testCases := []string{
+		"sum",
```

Comment:
[**For reviewers**] I haven't added tests for histograms for the metric to metric case. This is because, I couldn't think of a good example for these without implementing `conditions` (will be a follow-up). Use-cases like gauge to histograms OR histograms to exponential histograms will only make sense if we can add conditions on metric types or metric names. So, these will be added with the PR for conditions.